XLS-0066d-lending-protocol/README.md

```diff
@@ -0,0 +1,1428 @@
+
+$$
+totalDue = principalOutstanding + accruedInterest + prepaymentPenalty + ClosePaymentFee
+$$
+
+Accrued interest up to the point of early closure is calculated as follows:
+
+$$
+accruedInterest = principalOutstanding  \times periodicRate \times \frac{secondsSinceLastPayment}{paymentInterval}
+$$
+
+Finally, the Lender may charge a prepayment penalty for paying a loan early, which is calculated as follows:
+
+$$
+prepaymentPenalty = principalOutstanding \times closeInterestRate
+$$
+
+An early payment pays less interest than calculated when increasing the Vault value in the `LoanSet` transaction. Therefore, the Vault value (captured by `Vault.AssetTotal`) must be recalculated after an early payment.
+
+Assume a function `CurrentValue()` that returns `principalOutstanding` and `interestOutstanding` of the Loan. Furthermore, assume a function `ClosePayment()` that implements the Full Payment calculation. The function returns the total full payment due split into `principal` and `interest`.
+
+The value change for an early full repayment is calculated as follows:
+
+$$
+valueChange = (prepaymentPenalty) - (interestOutstanding - accruedInterest)
+$$
+
+Note that `valueChange <= 0` as an early repayment reduces the total value of the Loan.
+
+###### 3.2.5.1.5 Management Fee Calculations
+
+The `LoanBroker` Management fee is charged against the interest portion of the Loan and subtracted from the total Loan value at Loan creation. However, the fee is charged only during Loan payments. Early and Late payments change the total value of the Loan by decreasing or increasing the value of total interest. Therefore, when an early, late or an overpayment payment is made, the management fee must be updated.
+
+To update the management fee, we need to compute the new total management fee based on the new total interest after executing the early or late payment. Therefore, we need to capture the Loan value before the payment is made and the new value after the payment is made.
+
+For the calculation, assume the following variables:
+
+- Let $\mathcal{P}$ and $\mathcal{p}$ represent the total and outstanding Loan principal.
+- Let $\mathcal{I}$ and $\mathcal{i}$ represent the total and outstanding Loan interest computed from $\mathcal{P}$ and $\mathcal{p}$ respectively.
+- Let $\mathcal{V}$ and $\mathcal{v}$ represent the total and outstanding value of the Loan. $\mathcal{V} = \mathcal{P} + \mathcal{I}$ and $\mathcal{v} = \mathcal{p} + \mathcal{i}$.
+- Finally, let $\mathcal{m}$ represent the management fee rate of the Loan Broker.
+
+Assume $f(\mathcal{v})$ is a Loan payment, $f(\mathcal{v}) = \mathcal{v'}$, the new outstanding loan value is equal to the application of the payment transaction to the current outstanding value. Furthermore, assume $\mathcal{V} \xrightarrow{f(\mathcal{v})} \mathcal{V'}$, is the change in the Loan total value as the result of applying $f(\mathcal{v})$.
```

Comment:
`$\mathcal{V} \xrightarrow{f(\mathcal{v})} \mathcal{V'}$` rendered weird:
$\mathcal{V} \xrightarrow{f(\mathcal{v})} \mathcal{V'}$
I assume you want something like:
$\mathcal{V}$ $\xrightarrow{f(\mathcal{v})}$ $\mathcal{V'}$
But there's got to be a more elegant way to do it than what I did here: `$\mathcal{V}$ $\xrightarrow{f(\mathcal{v})}$ $\mathcal{V'}$`

Reply:
![image](https://github.com/user-attachments/assets/bca2aeda-d162-4949-a6c8-69d91e63795e)

Latex? Elegant?

Reply:
> Funny, how different Latex engines view things differently:

Oh, weird, so you probably don't even know why I'm complaining:

![image](https://github.com/user-attachments/assets/fdb063f3-06af-4e60-a95b-1392f98d6b43)

> Latex? Elegant? (I'll use your solution for now)

Ha! True!